modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -279,6 +322,12 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 		EntityField entityField = entityFieldsMap.get(
 			GetterUtil.getString(left));
 
+		if (_getDriverName().contains("HSQL")) {
+			String dateTimeString = right.toString();
+
+			right = _convertDateFormat(dateTimeString);
```

Comment:
Is it always going to be of date type? Shouldn't we be checking the type of this field?

Reply:
yes, you're right. The field should be checked the type